addons/im_livechat/controllers/main.py

```diff
@@ -74,6 +74,48 @@ def livechat_init(self, channel_id):
             'rule': rule,
         }
 
+    @http.route('/im_livechat/operator/<int:operator_id>/avatar',
+        type='http', auth="public", cors="*")
+    def livechat_operator_get_avatar(self, operator_id):
+        """ Custom route allowing to retrieve an operator's avatar.
+
+        This is done to bypass more complicated rules, notably 'website_published' when the website
+        module is installed.
+
+        Here, we assume that if you are a member of at least one im_livechat.channel, then it's ok
+        to make your avatar publicly available.
+
+        We also make the chatbot operator avatars publicly available. """
+
+        is_livechat_member = False
+        operator = request.env['res.partner'].sudo().browse(operator_id)
+        if operator.exists():
+            is_livechat_member = bool(request.env['im_livechat.channel'].sudo().search_count([
```

Comment:
Should we check that the user is online? Can you see the avatar of a user that is offline?

Reply:
Hello @mart-e 

The idea was to have a light solution in terms of checks.
If you are an operator of a livechat, we assume that you implicitly agree to share your profile picture.
(You have to actively click on a button to join the operators of a livechat channel).

Being online or offline should not matter I think.
Especially since it would be strange to refresh your conversation as a visitor the next day and not see the operator's avatar anymore because the operator is on holiday ^^